dipy/nn/histo_resdnn.py

```diff
@@ -0,0 +1,245 @@
+        res_add = Add()([x2, x4])
+        x5 = Dense(200, activation='relu')(res_add)
+        x6 = Dense(num_hidden)(x5)
+
+        self.model = Model(inputs=inputs, outputs=x6)
+
+    def load_model_weights(self, weights_path):
+        r"""
+        Load the model pre-training weights to use for the fitting.
+        Will not work if the declared SH_ORDER does not match the weights
+        expected input.
+
+        The weights for a sh_order of 8 can be obtained via the function:
+            get_fnames('histo_resdnn_weights').
+
+        Parameters
+        ----------
+        weights_path : str
+            Path to the file containing the weights (hdf5, saved by tensorflow)
+        """
+        try:
+            self.model.load_weights(weights_path)
+        except ValueError:
+            raise ValueError('Expected input for the provided model weights '
+                             'do not match the declared model ({})'
+                             .format(self.sh_size))
+
+    def predict(self, x_test):
+        r"""
+        Predict fODF (as SH) from input raw DWI signal (as SH)
+
+        Parameters
+        ----------
+        x_test : np.ndarray
+            Array of size (N, M) where M is
+            ``(sh_order + 1) * (sh_order + 2) / 2``.
+            N should not be too big as to limit memory usage.
+
+        Returns
+        -------
+        _ : np.ndarray (N, M)
+            Predicted fODF (as SH)
+        """
+
+        if x_test.shape[-1] != self.sh_size:
+            raise ValueError('Expected input for the provided model weights '
+                             'do not match the declared model ({})'
+                             .format(self.sh_size))
+
+        return self.model.predict(x_test)
+
+    def fit(self, data, gtab, mask=None):
+        """ Wrapper function to faciliate prediction of larger dataset.
+        The function will mask, normalize, split, predict and 're-assemble'
+        the data as a volume.
+
+        Parameters
+        ----------
+        data : np.ndarray
+            DWI signal in a 4D array
+        gtab : GradientTable class instance
+            The acquisition scheme matching the data (must contain at least
+            one b0)
+        mask : np.ndarray (optional)
+            Binary mask of the brain to avoid unnecessary computation and
+            unreliable prediction outside the brain.
+
+        Returns
+        -------
+        pred_sh_coef : np.ndarray (x, y, z, M)
+            Predicted fODF (as SH). The volume has matching shape to the input
+            data, but with ``(sh_order + 1) * (sh_order + 2) / 2`` as a last
+            dimension.
+
+        """
+        if mask is None:
+            logger.warning('Mask should be provided to accelerate '
+                           'computation, and because predictions are '
+                           'not reliable outside of the brain.')
+            mask = np.sum(data, axis=-1)
```

Comment:
I changed the description to add
```
Default: Compute prediction only for nonzero voxels (with at least
            one nonzero SH coefficient).
```
If there is noises my line will be the whole volume, if it is already skull-strip it's going to avoid computation where SH are all zeros.